Formula/ibex.rb

```diff
@@ -12,37 +24,82 @@ class Ibex < Formula
     sha256 "41327b6a0da9b8b2ed888c3c2c33d5f6ff061ea13eceb4b9e8b6e660049d624d" => :mountain_lion
   end
 
-  option "with-java", "Build Java bindings for Choco solver."
+  option "with-java", "Enable Java bindings for CHOCO solver."
+  option "with-affine", "Use Affine Arithmetic plugin"
+  option "with-ensta-robotics", "Use Contractors for robotics (SLAM) plugin"
+  option "with-ampl", "Use AMPL file loader plugin"
+  option "with-optim", "Use IbexOpt (Global optimization and System solving) plugin"
+  option "with-param-estim", "Use Parameter Estimation (enhanced Q-intersection algorithm) plugin"
+
+  option "with-debug", "Build Ibex with debug support and disable optimisations."
 
   depends_on :java => ["1.8+", :optional]
   depends_on "bison" => :build
   depends_on "flex" => :build
-  depends_on "pkg-config" => :build
+  depends_on "pkg-config"
+
+  if build.with?("java") && build.with?("ampl")
+    odie "Cannot set options --with-java and --with-ampl simultaneously for now."
+  end
 
   def install
     args = ["--prefix=#{prefix}"]
+    args << "--enable-shared"
+
+    args << "--with-debug" if build.with? "debug"
 
     if build.with? "java"
       args << "--with-jni"
-    else
-      args << "--enable-shared"
     end
 
+    if build.with? "affine"
+      args << "--with-affine"
+      opoo "LinearRelaxAffine class (from option --with-affine) will not be generated " \
+           "(requires --with-optim)" if build.without? "optim"
+    end
+
+    args << "--with-ensta-robotics" if build.with? "ensta-robotics"
+    args << "--with-ampl" if build.with? "ampl"
+    args << "--with-optim" if build.with? "optim"
+    args << "--with-param-estim" if build.with? "param-estim"
+
     system "./waf", "configure", *args
     system "./waf", "install"
 
-    cd "examples" do
-      cxxflags = "-frounding-math -ffloat-store -I#{include} -I#{include}/ibex"
-      libflags = "-L#{lib} -libex -lprim -lClp -lCoinUtils -lm"
-      system "make", "defaultsolver", "LIBS=#{libflags}", "CXXFLAGS=#{cxxflags}"
-    end
-
     pkgshare.install %w[examples benchs]
   end
 
   test do
     cp_r "#{pkgshare}/examples/.", testpath
+
+    # Helper function to make and then run a build target
+    def make_and_run(a)
+      system "make", a
+      system "./#{a}"
+    end
+
+    # Base Ibex examples
+    ohai "Building and running base Ibex examples"
+    open("symb01.txt", "w") { |f| f.puts "function f(x)\nreturn ((2*x,-x);(-x,3*x));\nend" }
```

Comment:
You can probably use either `(testpath/"symb01.txt").write` or `(testpath/"symb01.txt").append_lines` here.